plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/perspective/widgets/machine/panel/MachineTreeNode.java

```diff
@@ -48,8 +47,8 @@ public MachineTreeNode(@Assisted MachineTreeNode parent,
 
         boolean isMachine = data instanceof Machine;
 
-        id = isMachine ? ((MachineDto)data).getId() : ROOT;
-        name = isMachine ? ((MachineDto)data).getConfig().getName() : ROOT;
+        id = isMachine ? ((Machine)data).getId() : ROOT;
```

Comment:
Ugly unconditional type casting, I think it would be significantly better if we could check instance type here and throw warning/error/exception if that does not suit our logic instead of implicit type casting. Does that make sense for you?